### ISSUE TITLE
Don't require 'static

### ISSUE THREAD

**Issue by petersn** [Aleph-Alpha/ts-rs#116]
I see in the docs a todo labeled "don’t require 'static". I assume this is getting at how:
```
#[derive(ts_rs::TS)]
struct Foo<'a> {
    field: &'a str,
}
```
fails unless we add the constraint `'a: 'static`.

So far ts-rs is really great for helping keep my backend and frontend in sync, but not supporting non-`'static` lifetimes is a big limiter for my higher performance uses, where I really don't want copies in serde. Is there an ETA on this feature?

Thanks again for creating a great crate, by the way.

-Peter

**Comment by petersn** [comment]
Okay, I decided to jump in and see if I could hackily make this work. Here's my attempt:

https://github.com/Aleph-Alpha/ts-rs/pull/117/files

I'd love input if this is even remotely a reasonable way to get ts-rs to support non-`'static` lifetimes.

**Comment by escritorio-gustavo** [comment]
The docs are actully outdated on this. Lifetimes were fixed in #128
